agent/session_endpoint_test.go

```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/pascaldekloe/goe/verify"
 )
 
-func verifySession(t *testing.T, a *TestAgent, want structs.Session) {
+func verifySession(t *testing.T, r *retry.R, a *TestAgent, want structs.Session) {
```

Comment:
Seems strange to pass both of these now. I wonder if we could make retry.R implement the full testing.T interface and then use something like `mitchellh/go-testing-interface` as the type hint on helpers like this so we can pass either `t` or `r` and use them interchangeably...

But don't want to derail this PR too much. Maybe add a TODO comment here just so we might be prompted if we come across this again?

Reply:
@banks so the reason why I'm not removing `t *testing.T` is to keep the `t.Helper()`

However, after taking a closer look, t.Helper doesn't do its job when Retry is in use, so I'm just going to go ahead and remove `testing.T` and `t.Helper()`.